grails-app/services/org/pih/warehouse/core/DashboardService.groovy

```diff
@@ -554,6 +554,50 @@ class DashboardService {
         return reorderStock
     }
 
+    def getReorderReport(Location location) {
+        long startTime = System.currentTimeMillis()
+        ArrayList<Map> inventoryItems = getInventoryItems(location)
+
+        ArrayList<Map> reorderStock = inventoryItems.findAll { Map inventoryItem ->
+            def inventoryLevel = inventoryItem?.inventoryLevel as InventoryLevel
+
+            if (inventoryLevel?.binLocation != null) {
+                return false;
+            }
+
+            def quantityATP = inventoryItem.quantityAvailableToPromise as Number
+
+            def minQuantity = inventoryLevel?.minQuantity
+            if (minQuantity && quantityATP >= minQuantity) {
```

Comment:
Be careful. Please carefully consider the case where minQuantity = 0.If you intend the first expression to be minQuantity != null, that is not what you're getting and it could be short-circuiting on that expression. I don't have capacity for the mental gymnastics required here, but just know that it has bit me in the past.